### ISSUE TITLE
Adding and moving quotes in JS entered through source view

### ISSUE THREAD

**Issue by jaydiablo** [cheeaun/mooeditable#15]
When pasting in some javascript source code to the source view of mooeditable (can be done using the MooEditable.html demo file in this package) quotes get moved around, breaking the JS code pasted.

Example:

Load Demos/MooEditable/MooEditable.html in your browser (Firefox 3.6 Mac is what I've been testing with), open the source view, paste in the following Google Anayltics code:

```
<script type="text/javascript"><!--
google_ad_client = "pub-1234567789";
/* comment */
google_ad_slot = "1234567890";
google_ad_width = 300;
google_ad_height = 250;
//-->
</script>
<script type="text/javascript" src="http://pagead2.googlesyndication.com/pagead/show_ads.js">
</script>
```

Go back to wysiwyg mode, then click on the source icon again, notice now that the source for the JS pasted above now looks like:

```
<p><script type="text/javascript"><!--
google_ad_client = "pub-1234567789";
/* comment */
google_ad_slot =" "1234567890";
google_ad_width" =" 300;
google_ad_height" =" 250;
//--">
</script></p>
<p><script type="text/javascript" src="http://pagead2.googlesyndication.com/pagead/show_ads.js">
</script></p>
```

The extraneous paragraph tags aren't so much a problem as the moving of the quotes are (though the paragraph tags shouldn't be added either, IMO, since this was done in source view, not in wysiwyg view).

I'll look in the source some and see if I can patch this, but I'm not familiar with mooeditable's code base yet, so figured I'd post the issue first in case the pros out there know where to look. :)

**Comment by jaydiablo** [comment]
Quick look at the source and it seems the "Put quotes around unquoted attributes" regex in the cleanup method of MooEditable.js (line 774) is the culprit here.  It thinks the comment tags are an html element, and the = signs it sees in the JS code between the comment tags are html attributes so it inserts some quotes where it doesn't see any.  I was aboe to fix my test case here by changing the initial regex from:

```
/<[^>]*>/g
```

to:

```
/<[^!>]*>/g
```

Which won't match comments, though it may not match elements that it should if they contain any exclamation points in the attribute value (which I assume is legal html syntax).

**Comment by cheeaun** [comment]
Fixed in 5ad2c6339473e0775e1d1cf4ffd2783cb9cf6358
